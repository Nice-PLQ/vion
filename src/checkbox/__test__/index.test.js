import { mount } from '@vue/test-utils';
import Checkbox from '../checkbox.vue';

describe('Checkbox', () => {
  const wrapper = mount(Checkbox, {
    slots: {
      default: '<span>多选框</span>'
    },
    props: {
      size: '26px',
      checkedColor: '#67C23A'
    }
  });

  const childWrapper = wrapper.find('.v-checkbox__icon');

  test('选中', async () => {
    await wrapper.setProps({ checked: true });
    expect(childWrapper.classes()).toContain('v-checkbox_checked');
    expect(wrapper.element).toMatchSnapshot();
  });

  test('大小及选中后的背景色', () => {
    expect(childWrapper.element.style.fontSize).toBe('26px');
    expect(childWrapper.element.style.backgroundColor).toBe('rgb(103, 194, 58)');
  });

  test('方形的多选框', async () => {
    await wrapper.setProps({ shape: 'square' });
    expect(childWrapper.classes()).toContain('v-checkbox_square');
  });

  test('多选框的文本标签', () => {
    expect(wrapper.find('.v-checkbox__label').text()).toBe('多选框');
  });

  test('click & change事件触发', async () => {
    await wrapper.setProps({ checked: false });
    expect(wrapper.find('input[type="checkbox"]').element.checked).toBe(false);
    wrapper.trigger('click');
    expect(wrapper.emitted('click')).toBeTruthy();
    expect(wrapper.emitted('change')).toBeTruthy();
  });

  test('多选框禁用', async () => {
    const innerWrapper = mount(Checkbox, {
      props: {
        disabled: true
      }
    });
    await innerWrapper.trigger('click');
    expect(innerWrapper.emitted('click')).toBeFalsy();
  });
});
