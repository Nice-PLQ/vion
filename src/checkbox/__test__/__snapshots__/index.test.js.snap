// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox 选中 1`] = `
<div
  class="v-checkbox"
>
  <div
    class="v-checkbox__icon v-checkbox_checked"
    style="font-size: 26px; border-color: #67c23a; background-color: rgb(103, 194, 58);"
  >
    <input
      class="v-checkbox__input"
      type="checkbox"
    />
    <svg
      fill="currentColor"
      height="0.4166666666666667em"
      viewBox="0 0 28 20"
      width="0.5833333333333334em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <!-- eslint-disable max-len -->
      <path
        d="M27.5794 2.82843L24.751 0L9.90069 14.8503L2.82835 7.77794L-8.07047e-05 10.6064L8.4852 19.0916C8.82393 19.4304 9.25393 19.6222 9.69603 19.6671C10.2741 19.7265 10.873 19.5348 11.3159 19.0919L27.5794 2.82843Z"
        fill="currentColor"
      />
    </svg>
  </div>
  <span
    class="v-checkbox__label"
  >
    
    <span>
      多选框
    </span>
    
  </span>
</div>
`;
